src/librustc/ty/constness.rs

```diff
@@ -70,8 +70,11 @@ pub fn provide<'tcx>(providers: &mut Providers<'tcx>) {
         let hir_id = tcx.hir().as_local_hir_id(def_id)
                               .expect("Non-local call to local provider is_const_fn");
 
-        if let Some(fn_like) = FnLikeNode::from_node(tcx.hir().get_by_hir_id(hir_id)) {
+        let node = tcx.hir().get_by_hir_id(hir_id);
+        if let Some(fn_like) = FnLikeNode::from_node(node) {
             fn_like.constness() == hir::Constness::Const
+        } else if let hir::Node::Ctor(_) = node {
```

Comment:
`FnLikeNode` has a body method that's still used for now. It can probably be removed once AST borrowck is gone. While I don't think anyone else tries to get the `FnLikeNode` for constructors it doesn't seem to be the right thing for now.